src/docs/content/extras/canary.md

```diff
@@ -0,0 +1,246 @@
+ - Create a deployment with the below config and name it mushop-storefrontv2.
+   Note: We will use the same labels of storefront original.
+
+  ```shell--macos-linux
+    cat << EOF | kubectl apply -f -
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    metadata:
+      labels:
+        app.kubernetes.io/instance: mushop
+        app.kubernetes.io/name: storefront
+      name: mushop-storefrontv2
+      namespace: mushop
+    spec:
+      replicas: 1
+      selector:
+        matchLabels:
+          app: storefront
+          app.kubernetes.io/instance: mushop
+          app.kubernetes.io/name: storefront
+      template:
+        metadata:
+          labels:
+            app: storefront
+            app.kubernetes.io/instance: mushop
+            app.kubernetes.io/name: storefront
+            version: 2.1.3
+        spec:
+          containers:
+          - image: syd.ocir.io/ociateam/mushop/storefront:beta
+            imagePullPolicy: Always
+            name: storefront
+    EOF
+  ```
+
+
+## Create Istio resources
+
+- Deploy a Gateway resource
+
+  ```shell--macos-linux
+    cat << EOF | kubectl apply -f -
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: gateway
+      namespace: mushop
+    spec:
+      selector:
+        istio: ingressgateway
+      servers:
+        - port:
+            number: 80
+            name: http
+            protocol: HTTP
+          hosts:
+            - '*'
+    EOF
+    ```
+
+- Deploy a VirtualService and DestinationRules
+  
+  ```shell--macos-linux
+    cat <<EOF | kubectl apply -f -
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: edge
+      namespace: mushop
+    spec:
+      hosts:
+        - '*'
+      gateways:
+        - gateway
+      http:
+      - match:
+        - uri:
+            prefix: /api
+        route:
+        - destination:
+            host: mushop-api.mushop.svc.cluster.local
+      - match:
+        - uri:
+            prefix: /assets
+        rewrite:
+          uri: /
+        route:
+        - destination:
+            host: mushop-assets.mushop.svc.cluster.local
+      - route:
+        - destination:
+            host: mushop-storefront.mushop.svc.cluster.local
+            port:
+              number: 80
+            subset: original
+          weight: 50
+        - destination:
+            host: mushop-storefront.mushop.svc.cluster.local
+            port:
+              number: 80
+            subset: beta
+          weight: 50
+        EOF
+    ```
+
+   - Destination Rule:
+    
+    ```shell--macos-linux
+    cat <<EOF | k apply -f -
+    apiVersion: networking.istio.io/v1alpha3
+    kind: DestinationRule
+    metadata:
+      name: reviews-destination
+    spec:
+      host: mushop-storefront.mushop.svc.cluster.local
+      subsets:
+      - name: original
+        labels:
+          version: 2.1.2
+      - name: beta
+        labels:
+          version: 2.1.3
```

Comment:
I believe we're missing something in this strategy for `beta`. The beta image build would be something like `2.1.3-beta.1` and should trigger automatically with each of our builds. This would relate to #121 where it's possible to specify both targets in the `src/storefront/VERSION` file

Reply:
Including `2.1.3-beta.1` on the `src/storefront/VERSION` will trigger the auto creation of the image on the OCIR, but the regular services will still be using the 2.1.2 if do not change the manifests. I think this is the expected behavior, right?

Reply:
some clarifications:
1. syd.ocir.io/ociateam/mushop/storefront:beta is a temporary placeholder within the deployment.  I will need the image details once its built automatically so that i will amend the doc.

2. Also, the original storefront deployment had 2.1.2 label hence i added version=2.1.3 to new deployment + other labels , this is for istio routing. Not sure if this would have any relation to the way we build.

Reply:
Basically we need to implement #121 in order to automate build of both tags.

Reply:
We can build/push the beta image manually to test the updated doc

Reply:
@junior could you help build this image manually and push it to the repo. once done, please provide the image link to me, i can test it and amend the doc.

Reply:
I can do this for you tonight

Reply:
Thank you Matt 👍

Reply:
done:
-  https://iad.ocir.io/oracle/ateam/mushop-storefront:2.1.3
-  https://iad.ocir.io/oracle/ateam/mushop-storefront:2.1.3-beta.1

Reply:
Changes done.